mode/xml/xml.js

```diff
@@ -399,7 +399,7 @@ CodeMirror.defineMode("xml", function(editorConf, config_) {
     xmlCurrentContext: function(state) {
       var context = []
       for (var cx = state.context; cx; cx = cx.prev)
-        if (cx.tagName) context.push(cx.tagName)
+        context.push(cx.tagName)
```

Comment:
This changes the return value of this function, which looks like it will confuse the closetag and xml-hint addons. Why is this change needed?

Reply:
Right now what happens when you do something like

<{foo()}> or even <>, the bit here

https://github.com/codemirror/CodeMirror/blob/master/addon/edit/closetag.js#L131

causes context[context.length - 1] to be context[-1], which is undefined. So the autocomplete would have

<{foo()}></undefined>